config/base/derive/src/lib.rs

```diff
@@ -500,3 +500,336 @@ fn impl_configurable(ast: &DeriveInput) -> TokenStream {
+                        #[derive()]
+                    )
+                }
+                _ => {}
+            }
+        });
+    remove_attr_struct(&mut ast, "view");
+    ast.ident = format_ident!("{}View", ast.ident);
+    ast
+}
+
+fn gen_impl_from(original: &ViewInput, view: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput {
+        ident: original_ident,
+        ..
+    } = original;
+    let ViewInput {
+        generics,
+        ident: view_ident,
+        fields,
+        ..
+    } = view;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+    let field_idents = extract_field_idents(fields);
+
+    quote! {
+        impl #impl_generics core::convert::From<#original_ident> for #view_ident #ty_generics #where_clause {
+            fn from(config: #original_ident) -> Self {
+                let #original_ident {
+                    #(
+                        #field_idents,
+                    )*
+                    ..
+                } =  config;
+                Self {
+                    #(
+                        #field_idents: core::convert::From::<_>::from(#field_idents),
+                    )*
+                }
+            }
+        }
+    }
+}
+
+fn gen_impl_default(original: &ViewInput, view: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput {
+        ident: original_ident,
+        ..
+    } = original;
+    let ViewInput {
+        generics,
+        ident: view_ident,
+        ..
+    } = view;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    quote! {
+        impl #impl_generics core::default::Default for #view_ident #ty_generics #where_clause {
+            fn default() -> Self {
+                core::convert::From::<_>::from(<#original_ident as core::default::Default>::default())
+            }
+        }
+    }
+}
+
+fn gen_impl_has_view(original: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput {
+        generics,
+        ident: view_ident,
+        ..
+    } = original;
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    quote! {
+        impl #impl_generics iroha_config_base::view::HasView for #view_ident #ty_generics #where_clause {}
+    }
+}
+
+fn gen_assertions(view: &ViewInput) -> proc_macro2::TokenStream {
+    let ViewInput { fields, .. } = view;
+    let field_types = extract_field_types(fields);
+    let messages: Vec<String> = extract_field_idents(fields)
+        .iter()
+        .map(|ident| {
+            format!("Field `{ident}` has it's own view consider adding attribute #[view(into = ViewType)]")
+        })
+        .collect();
+    quote! {
+        /// Assert that every field of 'View' doesn't implement `HasView` trait
+        const _: () = {
+            use iroha_config_base::view::NoView;
+            #(
+                const _: () = assert!(!iroha_config_base::view::IsHasView::<#field_types>::IS_HAS_VIEW, #messages);
```

Comment:
Wait, is it basic [std assert](https://doc.rust-lang.org/std/macro.assert.html)? Then how does this work at compile time?

Reply:
I think we can add tests for this, like we have for `version` macro. That could be another issue, this one is already pretty big

Reply:
Yeah, we could use basic `assert!` in `const` context since [rust 1.57](https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html).